src/meta/processors/admin/DropSnapshotProcessor.cpp

```diff
@@ -35,34 +46,25 @@ void DropSnapshotProcessor::process(const cpp2::DropSnapshotReq& req) {
   }
   auto val = nebula::value(ret);
 
-  auto hosts = MetaKeyUtils::parseSnapshotHosts(val);
-  auto peersRet = NetworkUtils::toHosts(hosts);
-  if (!peersRet.ok()) {
+  auto hostsStr = MetaKeyUtils::parseSnapshotHosts(val);
+  auto hostsRet = NetworkUtils::toHosts(hostsStr);
+  if (!hostsRet.ok()) {
     LOG(INFO) << "Get checkpoint hosts error";
     handleErrorCode(nebula::cpp2::ErrorCode::E_SNAPSHOT_FAILURE);
     onFinished();
     return;
   }
-
-  auto peers = peersRet.value();
+  auto hosts = hostsRet.value();
   auto batchHolder = std::make_unique<kvstore::BatchHolder>();
-  auto dsRet = Snapshot::instance(kvstore_, client_)->dropSnapshot(snapshot, std::move(peers));
+  auto dsRet = Snapshot::instance(kvstore_, client_)->dropSnapshot(snapshot, std::move(hosts));
   if (dsRet != nebula::cpp2::ErrorCode::SUCCEEDED) {
-    LOG(INFO) << "Drop snapshot error on storage engine";
-    // Need update the snapshot status to invalid, maybe some storage engine
-    // drop done.
-    batchHolder->put(MetaKeyUtils::snapshotKey(snapshot),
```

Comment:
If we stay in INVALID, and drop again. Will it succeed or fail?

Reply:
It's a product behavior needed to be defined precisely:
1. If the user remove multi-times and still get an invalid, is it reasonable?
2. should we add an `--force` optional?

Anyway, I'll add an TODO and left it to be optimized the next time.

Reply:
👌